### PR TITLE
feat: support JSX and TSX in edge functions

### PR DESCRIPTION
**Pull request by ascorbic**
**Which problem is this pull request solving?**

Adds support for .jsx and .tsx entrypoints in edge functions

**Describe the solution you've chosen**

Deno includes out of the box support for jsx and tsx. However edge bundler doesn't recognise these files, so they can't be used for edge functions at the moment. This PR adds .tsx and .jsx to the list of allowed extensions. It allows things like this:

```tsx
import React from "https://esm.sh/react@18";
import { renderToReadableStream } from "https://esm.sh/v96/react-dom@18.2.0/deno/server.js";

export default async function handler(req: Request) {
  const url = new URL(req.url);
  const username = url.searchParams.get("username");

  if (!username) {
    url.searchParams.set("username", "netlify");
    return Response.redirect(url.toString(), 302);
  }

  const controller = new AbortController();
  let didError = false;
  try {
    const stream = await renderToReadableStream(
      <html>
        <title>Hello</title>
        <body>
          <h1>Hello {username}</h1>
        </body>
      </html>,
      {
        signal: controller.signal,
        onError(error: unknown) {
          didError = true;
          console.error(error);
        },
      },
    );

    return new Response(stream, {
      status: didError ? 500 : 200,
      headers: { "Content-Type": "text/html" },
    });
  } catch (error) {
    return new Response(
      "<!doctype html><p>Something went wrong</p>",
      {
        status: 500,
        headers: { "Content-Type": "text/html" },
      },
    );
  }
}
```

**Describe alternatives you've considered**

Importing a JSX file into a separate js or ts entrypoint

**Checklist**

Please add a `x` inside each checkbox:

- [x] I have read the [contribution guidelines](../blob/main/CONTRIBUTING.md).
- [x] The status checks are successful (continuous integration). Those can be seen below.

**A picture of a cute animal (not mandatory but encouraged)**
so im basically very smol

### CODE REVIEW

**Comment by ascorbic** [comment]
Aaah, it works locally because it's not in a directory. I'll update this

File: node/finder.ts [hunk 1]
```diff
@@ -4,13 +4,20 @@ import { basename, extname, join } from 'path'
...
+  const candidatePaths = [
+    `${name}.js`,
+    `index.js`,
+    `${name}.ts`,
+    `index.ts`,
+    `${name}.jsx`,
+    `index.jsx`,
+    `${name}.tsx`,
+    `index.tsx`,
+  ].map((filename) => join(directory, filename))
```
Comment (eduardoboucas) [inline node/finder.ts:20]: How do you feel about using `ALLOWED_EXTENSIONS` to generate all candidate paths, as opposed to enumerating them?

```suggestion
  const candidatePaths = [...ALLOWED_EXTENSIONS]
    .flatMap((extension) => [name + extension, `index${extension}`])
    .map((filename) => join(directory, filename))
```

Comment (ascorbic) [reply node/finder.ts]: Yeah, that is better

**Review by eduardoboucas — APPROVED** [review]
✨

**Pull request merged**